### PR TITLE
infra: Update openedu-builder to v0.5.1

### LINKED ISSUE #259: Update openedu-builder to the latest version
Updating `openedu-builder` to the latest version will simplify the config file and add automating link resolving for Markdown links.

### PR DESCRIPTION
**Pull request by VladNastase**
Closes #259

### CODE REVIEW

**Comment by VladNastase** [comment]
Regarding the checks, I'm not sure what kind of novel checkpatch wants as a commit message. As for the `Dockerfile`, I think it should be a separate issue/PR.

File: config.yaml [hunk 1]
```diff
@@ -38,198 +38,106 @@ docusaurus:
...
+              - Local I%2FO in Action: local-io-in-action.md
+              - Remote I%2FO: remote-io.md
+              - Networking 101: networking-101.md
+              - Client-Server Model: client-server-model.md
+              - Beyond Network Sockets: beyond-network-sockets.md
+              - File Mappings: file-mappings.md
+              - I%2FO Internals: io-internals.md
+              - Zero-Copy: zero-copy.md
+              - Asynchronous I%2FO: async-io.md
+              - I%2FO Multiplexing: io-multiplexing.md
```
Comment (teodutu) [inline config.yaml:115]: It's rather unfortunate that you have to URL-encode "I/O". Maybe it's better to just call it "IO".

File: config.yaml [hunk 2]
```diff
@@ -38,198 +38,106 @@ docusaurus:
...
-        - Application Interaction:
-            name: Application-Interaction
-            location: /build/embed_reveal/Application-Interaction
+          path: /build/embed_reveal
+          subsections:
+            - Software Stack: Software-Stack/Software-Stack.mdx
+            - Data: Data/Data.mdx
+            - Compute: Compute/Compute.mdx
+            - IO: IO/IO.mdx
+            - Application Interaction: Application-Interaction/Application-Interaction.mdx
```
Comment (teodutu) [inline config.yaml:49]: Does the builder create the `.mdx` files? The [makefile](https://github.com/open-education-hub/operating-systems/blob/master/content/common/makefile/slides.mk) simply creates a `.md` file.

**Review by teodutu — CHANGES_REQUESTED** [review]
My comments are questions rather than suggestions.

File: config.yaml (hunk 1 above)
Comment (VladNastase) [reply config.yaml:115]: I know, but in order to simplify the config, I had to give `/` the "meaning" of subfolder (see [here](https://open-education-hub.github.io/openedu-builder/plugins/docusaurus#structure), at the end). URL-encoding was really handy at the time, but I'll think of a way to make it more human-readable in a future version. Should I leave it like this or change it to "IO"?

File: config.yaml (hunk 2 above)
Comment (VladNastase) [reply config.yaml:49]: [Yes](https://github.com/open-education-hub/operating-systems/blob/bf54209d3c09803b914a0ddf517929ac8d7f9ad2/config.yaml#L27), those are from the output of the `reveal_embed` plugin.

File: config.yaml (hunk 1 above)
Comment (teodutu) [reply config.yaml:115]: Chnage it to IO. Otherwise it looks weird and forced.

Comment (VladNastase) [reply config.yaml]: Done

**Review by teodutu — APPROVED** [review]
Amen :orthodox_cross:.

**Comment by gabrielmocanu** [comment]
@VladNastase 

Please make an issue regarding Hadolint errors and then merge it when you feel confident.

**Comment by gabrielmocanu** [comment]
@VladNastase  Any reasons this is not merged yet?

**Comment by VladNastase** [comment]
Yes, I wanted to merge this outside of lab hours because it changes some URLs, but kept forgetting. Thanks for reminding me, I'll merge this now.

**Pull request merged**